### DISCUSSION TITLE
Aves detects only some files out of all of them

### DISCUSSION THREAD

**Discussion by N1N74** [deckerst/aves#1014] [General]
Hola, so it turns out that aves is detecting only 1053 files out of total 1126 in a particular folder within DCIM. I wanted to document the bug before clearing data and restarting. Anyway, here is fossify gallery and my file explorer detecting all the files. 

![image](https://github.com/deckerst/aves/assets/39383296/0eb4a662-29a8-41f3-bcda-1cc13e86fa22)
![image](https://github.com/deckerst/aves/assets/39383296/d1142a68-38c2-4f05-9743-8d599daa2df5)

And here is my Aves Libre f-droid install that only detects 1053 of them:
![image](https://github.com/deckerst/aves/assets/39383296/91ad7d01-03ee-4746-bb7a-aad3b9a24da9)

Aves libre instantly recognizes new files I move into it somehow so it still has access to view the folder but refuses to recognize some files like the one ending in WA0031.
![image](https://github.com/deckerst/aves/assets/39383296/c4255c84-f8fa-4ca3-8a39-6fcb556da3c6)
![image](https://github.com/deckerst/aves/assets/39383296/d161eb7f-6112-4452-884d-76abfed3fd6e)
PS I have tried clearing cache and restarting app multiple times, even tried force stopping.

**Comment by deckerst** [comment]
The files that you identified missing in Aves, can you open them in Aves from your file manager? Are they displayed correctly in that case?

And if you move one of these missing file to another directory, does Aves see it in its new location?

**Reply by N1N74** [reply]
I opened through file explorer and aves did open and play it properly as well. 

Checking the number count on the album afterwards, it's remained the same and the file isn't shown on search as usual. Here's what its info looks like, it doesn't say "pictures" as it should:
![image](https://github.com/deckerst/aves/assets/39383296/293761d0-ef2a-4c95-baa9-4ee3858bbf84)

**Reply by N1N74** [reply]
If I copied the problem file to an album using [files](https://f-droid.org/en/packages/com.github.axet.filemanager/) app, fossify recognizes whereas aves libre fails to do so, if the same was copied using fossify, both aves and fossify recognize. 

Actually if I create an empty folder with a file moved using [files](https://f-droid.org/en/packages/com.github.axet.filemanager/) app, both fossify and aves cannot recognize the new folder. Honestly mind-blowing stuff, this failure even persists on copying a previously recognized image to the new folder using the same app. 

I also think there is some weird cache/update issue as well. I renamed a previously recognized file using the files app and Aves didnt immediately update it like fossify did. Amazingly aves still played the video like it was never renamed but upon restarting app, the file name remains the same and is unplayable. Here is what the info shows:

![image](https://github.com/deckerst/aves/assets/39383296/1ed42248-7c57-4bf0-9915-1848eefb02e9)

**Reply by N1N74** [reply]
WOAH, what the hell, aves just recognized all the files suddenly. I didn't even do anything. The album now says 1126 files. I guess this is still a concern because of those other behaviors. iirc something similar happened during testing to one album when aves suddenly recognized all the files in it when I renamed just one file. 😕 Sorry abt this, if I can provide logs or need help testing something to navigate this I'd love to.

**Comment by deckerst** [comment]
As I understand it, some of the apps you use (the `files` app, maybe others) do not properly update the Android Media Store when handling media files.

Aves lists media as known by the system. Other galleries may choose to also scan directories to find the media files themselves, hence the discrepancy.

In my opinion, it's better to fix/use better file managers than requiring all the other apps to do manual file scanning, as it shouldn't be necessary. The Android Media Store is already here for that.

**Reply by N1N74** [reply]
I didn't suspect the file manager initially, but alright seems fair. Kind of an awkward situation though lol. 

Aves also can't copy, move or rename files from downloads folder because google in their infinite wisdom, decided to not allow apps access to downloads folder whereas fossify gallery can do that. I think these kinds of things hit the reliability of the app so, if there's ever any interest in opting for the alternative method of accessing files, imho that would turn out to be most helpful. ^^

**Reply by deckerst** [reply]
Aves can handle media in the `Downloads` directory, but it doesn't use the same API. Instead of asking for write access to a folder, it requests access to the media files as needed. The prompts can be tedious, so users can grant "media management" permission to skip the prompts. It's in Aves privacy settings (or you can also toggle it via Android app settings).

**Reply by N1N74** [reply]
Woah, sweet, I tried it and it worked! Thanks ^^
Now, I guess I'll try to find a foss file browser that also supports SAF and one that properly works too.
